hooks/useFetch.ts

```diff
@@ -0,0 +1,61 @@
+import { create } from "zustand";
+
+interface APIState {
+  data: any;
+  error: Error | null;
+  isLoading: boolean;
+}
+
+interface State {
+  wikiEntry: APIState;
+  updateWiki: APIState;
+  fetch: (route: keyof State, params: any) => Promise<void>;
+  alterWikiEntry: (data: string | undefined) => void;
+}
+
+export const useFetch = create<State>((set) => ({
+  wikiEntry: {
+    data: null,
+    error: null,
+    isLoading: false,
+  },
+  updateWiki: {
+    data: null,
+    error: null,
+    isLoading: false,
+  },
+  fetch: async (route, params) => {
+    set((state) => ({
+      [route]: { ...state[route], isLoading: true, error: null },
```

Comment:
teh fuck is happening here with `...state[route]` ? :D

Reply:
the route slugs also serve as key for fetch state.